### PR TITLE
feat: add a step to initialize git repo and create github repo

### PR DESCRIPTION
**Pull request by dyc3**
This adds a very quick and dirty script to automate creating a github repo and pushing the code to that repo.

### CODE REVIEW

**Comment by nhedger** [comment]
You seem to have gitignored the script.

**Comment by nhedger** [comment]
How about we prompt the user during the interactive session and ask him if he wants to publish the repo? If he agrees, we create the repo and push the initial code. He's then left with a git repo that already has a remote setup and can commit and push any subsequent code.

**Comment by dyc3** [comment]
> You seem to have gitignored the script.

:facepalm: I was thinking that it would just ignore the scripts in the generated repo lol

**Comment by nhedger** [comment]
> > You seem to have gitignored the script.
> 
> 🤦 I was thinking that it would just ignore the scripts in the generated repo lol

How other generators usually solve this, is to create a `_gitignore` file and rename it during scaffolding.

File: src/index.ts [hunk 1]
```diff
@@ -109,6 +159,21 @@ async function init() {
...
+		return true;
+	} catch (error) {
+		return false;
+	}
+}
+
+function initRepoAndCommit() {
+	execSync("git init", { stdio: "inherit" });
+	execSync("git add .", { stdio: "inherit" });
+	execSync('git commit -m "Initial commit"', { stdio: "inherit" });
```
Comment (nhedger) [inline src/index.ts]: I think maybe it looks better if we hide the output of these commands. After all, we're not committing anything the user has typed, so I thinks it's ok if it's a little less explicit.

```suggestion
	execSync("git init");
	execSync("git add .");
	execSync('git commit -m "Initial commit"');
```

**Review by nhedger — APPROVED** [review]

**Review by nhedger — APPROVED** [review]

**Comment by nhedger** [comment]
Thanks @dyc3 !

**Pull request merged**